tests/test_shield_privacy_prefs.py

```diff
@@ -14,14 +15,105 @@ def create_ping_rdd(sc, payload):
 
 def create_row(overrides):
     keys = ["client_id", "branch", "event", "originDomain", "breakage",
-            "notes", "study"]
+            "notes", "study", "study_state"]
 
-    overrides['study'] = '@shield-study-privacy'
+    # simulate transforuming study_name into study field
+    if 'study_name' in overrides:
+        del overrides['study_name']
+        overrides['study'] = '@shield-study-privacy'
     overrides['client_id'] = 'aa'
 
     return {key: overrides.get(key, None) for key in keys}
 
 
+def test_study_running(row_to_dict, spark_context):
```

Comment:
I still think these 4 tests are redundant. The only field that changes is `study_state`, but there is nothing that checks the validity of `study_state`. A better test would be an invalid or null `study_state`, plus a single valid case.